F1/F1/Login/LoginViewController.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  LoginViewController.swift
+//  F1
+//
+//  Created by Kaitlyn Sookdhew on 2024/04/10.
+//
+
+import UIKit
+
+class LoginViewController: UIViewController {
+
+    @IBOutlet weak var username: UITextField!
+    @IBOutlet weak var password: UITextField!
+    @IBOutlet weak var errorMsg: UILabel!
+    private lazy var viewModel = LoginViewModel()
+
+    override func viewDidLoad() {
+        errorMsg.isHidden = true
+        super.viewDidLoad()
+    }
+
+    @IBAction func loginAction(_ sender: Any) {
+        if viewModel.validDetails(givenUsername: username.text ?? "", givenPassword: password.text ?? "") {
```

Comment:
please move this logic to view model and use delegate methods

Reply:
marks